File: platforms/ios/SparrowInvest/Components/Dashboard/MarketOverviewCard.swift
//
//  MarketOverviewCard.swift
//  SparrowInvest
//
//  Market indices overview card
//

import SwiftUI

struct MarketOverviewCard: View {
    let marketOverview: MarketOverview

    var body: some View {
        VStack(alignment: .leading, spacing: 16) {
            // Header
            HStack {
                Text("Market Overview")
                    .font(.system(size: 16, weight: .semibold))
                    .foregroundColor(AppTheme.textPrimary)

                Spacer()

                // Market Status Badge
                HStack(spacing: 6) {
                    Circle()
                        .fill(marketOverview.status.color)
                        .frame(width: 8, height: 8)
                    Text(marketOverview.status.rawValue)
                        .font(.system(size: 12, weight: .medium))
                        .foregroundColor(marketOverview.status.color)
                }
                .padding(.horizontal, 10)
                .padding(.vertical, 5)
                .background(
                    Capsule()
                        .fill(marketOverview.status.color.opacity(0.1))
                )
            }

            // Primary Indices
            HStack(spacing: 16) {
                ForEach(marketOverview.primaryIndices) { index in
                    MarketIndexTile(index: index)
                }
            }

            // Other Indices (if any)
            if !marketOverview.otherIndices.isEmpty {
                ScrollView(.horizontal, showsIndicators: false) {
                    HStack(spacing: 12) {
                        ForEach(marketOverview.otherIndices) { index in
                            CompactIndexTile(index: index)
                        }
                    }
                }
            }

            // Last Updated
            HStack {
                Image(systemName: "clock")
                    .font(.system(size: 11))
                Text("Updated \(formatLastUpdated(marketOverview.lastUpdated))")
                    .font(.system(size: 11, weight: .medium))
            }
            .foregroundColor(AppTheme.textTertiary)
        }
        .padding(16)
        .background(
            RoundedRectangle(cornerRadius: 16)
                .fill(AppTheme.cardBackground)
                .shadow(color: AppTheme.cardShadow, radius: 8, x: 0, y: 2)
        )
    }

    private func formatLastUpdated(_ date: Date) -> String {
        let formatter = RelativeDateTimeFormatter()
        formatter.unitsStyle = .abbreviated
        return formatter.localizedString(for: date, relativeTo: Date())
    }
}

struct MarketIndexTile: View {
    let index: MarketIndex

    var body: some View {
        VStack(alignment: .leading, spacing: 8) {
            Text(index.name)
                .font(.system(size: 12, weight: .medium))
                .foregroundColor(AppTheme.textSecondary)

            Text(index.formattedValue)
                .font(.system(size: 18, weight: .bold))
                .foregroundColor(AppTheme.textPrimary)

            HStack(spacing: 4) {
                Image(systemName: index.isPositive ? "arrow.up" : "arrow.down")
                    .font(.system(size: 10, weight: .bold))
                Text(index.formattedChange)
                    .font(.system(size: 12, weight: .medium))
            }
            .foregroundColor(index.isPositive ? AppTheme.success : AppTheme.error)
        }
        .frame(maxWidth: .infinity, alignment: .leading)
        .padding(12)
        .background(
            RoundedRectangle(cornerRadius: 12)
                .fill(index.isPositive ? AppTheme.success.opacity(0.08) : AppTheme.error.opacity(0.08))
        )
    }
}

struct CompactIndexTile: View {
    let index: MarketIndex

    var body: some View {
        VStack(alignment: .leading, spacing: 4) {
            Text(index.name)
                .font(.system(size: 11, weight: .medium))
                .foregroundColor(AppTheme.textSecondary)

            HStack(spacing: 4) {
                Text(String(format: "%.0f", index.value))
                    .font(.system(size: 13, weight: .semibold))
                    .foregroundColor(AppTheme.textPrimary)

                Text("\(index.isPositive ? "+" : "")\(String(format: "%.1f", index.changePercentage))%")
                    .font(.system(size: 11, weight: .medium))
                    .foregroundColor(index.isPositive ? AppTheme.success : AppTheme.error)
            }
        }
        .padding(10)
        .background(
            RoundedRectangle(cornerRadius: 10)
                .fill(AppTheme.chipBackground)
        )
    }
}

#Preview {
    MarketOverviewCard(marketOverview: .empty)
        .padding()
}

File: platforms/android-consumer/app/src/main/java/com/sparrowinvest/app/ui/components/MarketOverviewCard.kt
package com.sparrowinvest.app.ui.components

import androidx.compose.foundation.background
import androidx.compose.foundation.horizontalScroll
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.AccessTime
import androidx.compose.material.icons.filled.ArrowDropDown
import androidx.compose.material.icons.filled.ArrowDropUp
import androidx.compose.material3.Icon
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.unit.dp
import com.sparrowinvest.app.data.model.MarketIndex
import com.sparrowinvest.app.data.model.MarketOverview
import com.sparrowinvest.app.data.model.MarketStatus
import com.sparrowinvest.app.ui.theme.CornerRadius
import com.sparrowinvest.app.ui.theme.Error
import com.sparrowinvest.app.ui.theme.LocalIsDarkTheme
import com.sparrowinvest.app.ui.theme.Spacing
import com.sparrowinvest.app.ui.theme.Success
import com.sparrowinvest.app.ui.theme.Warning
import java.util.Locale

@Composable
fun MarketOverviewCard(
    marketOverview: MarketOverview,
    modifier: Modifier = Modifier
) {
    GlassCard(modifier = modifier) {
        Column(modifier = Modifier.fillMaxWidth()) {
            // Header: Title + Status Badge
            Row(
                modifier = Modifier.fillMaxWidth(),
                horizontalArrangement = Arrangement.SpaceBetween,
                verticalAlignment = Alignment.CenterVertically
            ) {
                Text(
                    text = "Market Overview",
                    style = MaterialTheme.typography.titleSmall,
                    color = MaterialTheme.colorScheme.onSurface
                )

                MarketStatusBadge(status = marketOverview.statusEnum)
            }

            Spacer(modifier = Modifier.height(Spacing.medium))

            // Primary Indices: Row of 2 tiles
            val primaryIndices = marketOverview.primaryIndices
            if (primaryIndices.isNotEmpty()) {
                Row(
                    modifier = Modifier.fillMaxWidth(),
                    horizontalArrangement = Arrangement.spacedBy(Spacing.compact)
                ) {
                    primaryIndices.forEach { index ->
                        PrimaryIndexTile(
                            index = index,
                            modifier = Modifier.weight(1f)
                        )
                    }
                    // Fill remaining space if only one primary index
                    if (primaryIndices.size == 1) {
                        Spacer(modifier = Modifier.weight(1f))
                    }
                }
            }

            // Other Indices: Horizontal scroll
            val otherIndices = marketOverview.otherIndices
            if (otherIndices.isNotEmpty()) {
                Spacer(modifier = Modifier.height(Spacing.compact))

                Row(
                    modifier = Modifier
                        .fillMaxWidth()
                        .horizontalScroll(rememberScrollState()),
                    horizontalArrangement = Arrangement.spacedBy(Spacing.small)
                ) {
                    otherIndices.forEach { index ->
                        CompactIndexTile(index = index)
                    }
                }
            }

            // Last Updated
            if (marketOverview.lastUpdated.isNotEmpty()) {
                Spacer(modifier = Modifier.height(Spacing.compact))

                Row(verticalAlignment = Alignment.CenterVertically) {
                    Icon(
                        imageVector = Icons.Default.AccessTime,
                        contentDescription = null,
                        tint = MaterialTheme.colorScheme.onSurfaceVariant.copy(alpha = 0.6f),
                        modifier = Modifier.size(14.dp)
                    )
                    Spacer(modifier = Modifier.width(Spacing.micro))
                    Text(
                        text = marketOverview.lastUpdated,
                        style = MaterialTheme.typography.labelSmall,
                        color = MaterialTheme.colorScheme.onSurfaceVariant.copy(alpha = 0.6f)
                    )
                }
            }
        }
    }
}

@Composable
private fun MarketStatusBadge(status: MarketStatus) {
    val statusColor = when (status) {
        MarketStatus.OPEN -> Success
        MarketStatus.CLOSED -> Error
        MarketStatus.PRE_OPEN -> Warning
        MarketStatus.POST_CLOSE -> Warning
    }

    Row(
        modifier = Modifier
            .clip(RoundedCornerShape(50))
            .background(statusColor.copy(alpha = 0.12f))
            .padding(horizontal = Spacing.small, vertical = Spacing.micro),
        verticalAlignment = Alignment.CenterVertically
    ) {
        Box(
            modifier = Modifier
                .size(6.dp)
                .clip(CircleShape)
                .background(statusColor)
        )
        Spacer(modifier = Modifier.width(Spacing.micro))
        Text(
            text = status.displayName,
            style = MaterialTheme.typography.labelSmall,
            color = statusColor
        )
    }
}

@Composable
private fun PrimaryIndexTile(
    index: MarketIndex,
    modifier: Modifier = Modifier
) {
    val isDark = LocalIsDarkTheme.current
    val tintColor = if (index.isPositive) Success else Error
    val bgAlpha = if (isDark) 0.10f else 0.06f

    Column(
        modifier = modifier
            .clip(RoundedCornerShape(CornerRadius.medium))
            .background(tintColor.copy(alpha = bgAlpha))
            .padding(Spacing.compact)
    ) {
        Text(
            text = index.name,
            style = MaterialTheme.typography.labelSmall,
            color = MaterialTheme.colorScheme.onSurfaceVariant
        )

        Spacer(modifier = Modifier.height(Spacing.micro))

        Text(
            text = index.formattedValue,
            style = MaterialTheme.typography.titleMedium,
            color = MaterialTheme.colorScheme.onSurface
        )

        Spacer(modifier = Modifier.height(Spacing.micro))

        Row(verticalAlignment = Alignment.CenterVertically) {
            Icon(
                imageVector = if (index.isPositive) Icons.Default.ArrowDropUp else Icons.Default.ArrowDropDown,
                contentDescription = null,
                tint = tintColor,
                modifier = Modifier.size(18.dp)
            )
            Text(
                text = index.formattedChange,
                style = MaterialTheme.typography.labelSmall,
                color = tintColor
            )
        }
    }
}

@Composable
private fun CompactIndexTile(
    index: MarketIndex,
    modifier: Modifier = Modifier
) {
    val isDark = LocalIsDarkTheme.current
    val tintColor = if (index.isPositive) Success else Error

    Column(
        modifier = modifier
            .clip(RoundedCornerShape(CornerRadius.small))
            .background(
                if (isDark) Color.White.copy(alpha = 0.06f)
                else MaterialTheme.colorScheme.surfaceVariant
            )
            .padding(horizontal = Spacing.compact, vertical = Spacing.small)
    ) {
        Text(
            text = index.name,
            style = MaterialTheme.typography.labelSmall,
            color = MaterialTheme.colorScheme.onSurfaceVariant
        )

        Spacer(modifier = Modifier.height(Spacing.micro))

        Text(
            text = index.formattedValue,
            style = MaterialTheme.typography.bodySmall,
            color = MaterialTheme.colorScheme.onSurface
        )

        Spacer(modifier = Modifier.height(2.dp))

        Text(
            text = String.format(
                Locale.US,
                "%s%.2f%%",
                if (index.changePercentage >= 0) "+" else "",
                index.changePercentage
            ),
            style = MaterialTheme.typography.labelSmall,
            color = tintColor
        )
    }
}
